core/src/local/state.rs

```diff
@@ -1,18 +1,17 @@
 use ::*;
 use std::collections::HashMap;
-use std::hash::Hash;
 use std::rc::Rc;
 
 use local::keyed_state::KeyedState;
 use local::single_state::SingleState;
 
-pub struct State<T: Hash + Eq + Clone + 'static> {
-    state: Vec<SingleState<T>>,
+pub struct State {
+    state: Vec<SingleState>,
     by_tag: HashMap<Tag, usize>,
     rows: usize,
 }
 
-impl<T: Hash + Eq + Clone + 'static> Default for State<T> {
+impl Default for State {
```

Comment:
We could actually `#[derive(Default)]` for State now that the bounds don't get in the way

Reply:
Nice! Fixed in https://github.com/mit-pdos/distributary/pull/64/commits/59fd0c23f8e52b87e87c30a78e5184d0b8155d89.